src/main/java/org/springframework/integration/aws/inbound/kinesis/KclMessageDrivenChannelAdapter.java

```diff
@@ -288,47 +317,88 @@ public void processRecords(List<Record> records, IRecordProcessorCheckpointer ch
 			if (logger.isDebugEnabled()) {
 				logger.debug("Processing " + records.size() + " records from " + this.shardId);
 			}
-			for (Record record : records) {
+
+			if (ListenerMode.record.equals(KclMessageDrivenChannelAdapter.this.listenerMode)) {
+				for (Record record : records) {
+					try {
+						processSingleRecord(record, checkpointer);
+					}
+					catch (Throwable t) {
```

Comment:
@artembilan I kept the `try..catch(Throwable)` from the existing code but I'm not sure how useful it is. Do we want to catch this exception and just log a warning? Could be perfectly fine though, just wondering..

Reply:
Let's wrap everything into `try...finally() { attributesHolder.remove(); }` without any logs!
This way we will have the same logic like in the `KinesisMessageDrivenChannelAdapter`.
All other exceptions rather than send would be logged by the source client. I assume it is fully OK from our side.